master/internal/api_trials.go

```diff
@@ -68,36 +49,94 @@ var (
 )
 
 // TrialLogBackend is an interface trial log backends, such as elastic or postgres,
-// must support to provide the features surfaced in API.
+// must support to provide the features surfaced in API. This is deprecated, note it
+// no longer supports adding logs in favor of unified logs.
 type TrialLogBackend interface {
 	TrialLogs(
 		trialID, limit int, filters []api.Filter, order apiv1.OrderBy, state interface{},
 	) ([]*model.TrialLog, interface{}, error)
-	AddTrialLogs([]*model.TrialLog) error
 	TrialLogsCount(trialID int, filters []api.Filter) (int, error)
 	TrialLogsFields(trialID int) (*apiv1.TrialLogsFieldsResponse, error)
 	DeleteTrialLogs(trialIDs []int) error
 }
 
 func (a *apiServer) TrialLogs(
-	req *apiv1.TrialLogsRequest, resp apiv1.Determined_TrialLogsServer) error {
+	req *apiv1.TrialLogsRequest, resp apiv1.Determined_TrialLogsServer,
+) error {
+	var taskID model.TaskID
+	switch t, err := a.m.db.TrialByID(int(req.TrialId)); {
+	case errors.Is(err, sql.ErrNoRows), errors.Is(err, db.ErrNotFound):
+		return trialNotFound
+	case err != nil:
+		return err
+	default:
+		taskID = t.TaskID
+	}
+
+	ctx, cancel := context.WithCancel(resp.Context())
+	defer cancel()
+
+	res := make(chan api.BatchResult, 1)
+	switch t, err := a.m.db.TaskByID(taskID); {
+	case errors.Is(err, sql.ErrNoRows), t.LogVersion == 0:
```

Comment:
I think it would be a bit more straightforward if the condition were `if t, err := a.m.db.TaskByID(taskID); errors.Is(err, sql.ErrNoRows) || t.LogVersion == 0`. (For a while there, I got confused into thinking the equivalent was `&&` rather than `||`.)

Reply:
i kept it as a switch but mostly so i could make it explicit that every task log version is properly handed.